src/engine/plugins/trip.cpp

```diff
@@ -187,8 +211,76 @@ Status TripPlugin::HandleRequest(const std::shared_ptr<const datafacade::BaseDat
     BOOST_ASSERT_MSG(result_table.size() == number_of_locations * number_of_locations,
                      "Distance Table has wrong size");
 
+    // The following code manipulates the table and produces the new table for
+    // Trip with Fixed Start and End (TFSE)
+
+    // Original Table
+    //   a  b  c  d  e
+    // a 0  15 36 34 30
+    // b 15 0  25 30 34
+    // c 36 25 0  18 32
+    // d 34 30 18 0  15
+    // e 30 34 32 15 0
+
+    // New Table
+    //   a        b         c        d         e
+    // a 0        15        10000    34        30
+    // b 10000    0         25       30        34
+    // c 0        10000     0        10000     10000
+    // d 10000    30        18       0         15
+    // e 10000    34        32       15        0
+
+    if (parameters.source > -1 && parameters.destination > -1)
+    {
+        // parameters.source column
+        for (NodeID i = 0; i < result_table.GetNumberOfNodes(); i++) {
+            if (i == (NodeID)parameters.source) continue;
+            result_table.InvalidateRoute(i, (NodeID)parameters.source);
+        }
+
+        // parameters.destination row
+        for (NodeID i = 0; i <result_table.GetNumberOfNodes(); i++) {
+            if (i == (NodeID)parameters.destination) continue;
+            result_table.ShortcutRoute((NodeID)parameters.destination, i);
+        }
+
+        result_table.ShortcutRoute(parameters.destination, parameters.source);
+        result_table.InvalidateRoute(parameters.source, parameters.destination);
+    }
+
     // get scc components
-    SCC_Component scc = SplitUnaccessibleLocations(number_of_locations, result_table);
+    SCC_Component scc = SplitUnaccessibleLocations(result_table.GetNumberOfNodes(), result_table);
+
+    if (parameters.source > -1 && parameters.destination > -1) // check if fixed start and end
```

Comment:
later you're checking this to set  `bool roundtrip`. you could also set the boolean already here and use it.